addons/hr_holidays/tests/test_accrual_allocations.py

```diff
@@ -4015,24 +4015,248 @@ def test_accrual_allocation_date_in_the_future(self):
         allocation_data = leave_type.get_allocation_data(self.employee_emp, '2031-09-01')
         self.assertEqual(allocation_data[self.employee_emp][0][1]['virtual_remaining_leaves'], 23, "The carryover expired after 6 month so the remaining leaves should be 23")
 
-    def test_accrual_plan_cleared_when_switch_to_regular(self):
-        accrual_plan = self.env['hr.leave.accrual.plan'].with_context(tracking_disable=True).create({
+    def _form_get_allocation_sample(self, leave_type, date_from, accrual_plan=None, number_of_days=None, date_to=False):
+        with Form(self.env['hr.leave.allocation'], 'hr_holidays.hr_leave_allocation_view_form_manager') as f:
+            f.name = 'Accrual allocation for employee'
+            if accrual_plan:
+                f.allocation_type = 'accrual'
+                f.accrual_plan_id = accrual_plan
+            else:
+                f.allocation_type = 'regular'
+            f.employee_id = self.employee_emp
+            f.holiday_status_id = leave_type
+            f.date_from = date_from
+            if date_to:
+                f.date_to = date_to
+            if number_of_days is not None:
+                f.number_of_days_display = number_of_days
+        return f.record
+
+    def _get_leave_type_day_sample0(self):
+        return self.env['hr.leave.type'].create({
+            'name': 'Test Leave Type',
+            'time_type': 'leave',
+            'requires_allocation': 'yes',
+            'allocation_validation_type': 'no_validation',
+            'request_unit': 'day',
+        })
+
+    def _get_accrual_plan_sample0(self, accrued_days, accrued_gain_time, max_leaves=0):
```

Comment:
why do we need a 0 here? The method naming is a bit weird globally, maybe standardise the different methods with a name like `_create_test_<model name>` and put all of those inside `common` so they're not only usable here